### PR TITLE
Tests for Region middleware

### PR DESCRIPTION
**Pull request by rebecca-caktus**
* Create `SetRegion` view to encapsulate logic for setting the active region.
* Modularize `UserRegionsMiddleware`.
* Tests for `UserRegionsMiddleware`, `SetRegion`, and `ToggleSubregions`.

### CODE REVIEW

File: tracpro/groups/tests/test_middleware.py [hunk 1]
```diff
@@ -1,62 +1,179 @@
...
+        self.assertTrue(request.include_subregions)
+
+    def test_include_subregions__yes(self):
+        """include_subregions should be retrieved from the session."""
+        request = self.get_request(session={'include_subregions': True})
+        self.middleware.set_include_subregions(request)
+        self.assertTrue(request.include_subregions)
+
+    def test_include_subregions__no(self):
+        """include_subregions should be retrieved from the session."""
```
Comment (emullaney) [inline tracpro/groups/tests/test_middleware.py]: comment needs to be updated?

Comment (rebecca-caktus) [inline tracpro/groups/tests/test_middleware.py]: To be more specific about `True` vs `False`?

File: tracpro/groups/tests/test_views.py [hunk 2]
```diff
@@ -14,6 +14,177 @@
...
+    def test_set_region(self):
+        """Set region_id variable in the session."""
+        response = self.set_region({'region': self.region.pk})
+        self.assertRedirects(
+            response, reverse('home.home'), self.org.subdomain,
+            fetch_redirect_response=False)
+        self.assertEqual(self.client.session[self.session_key], str(self.region.pk))
+
+    def test_next_invalid(self):
+        """Should not redirect to an invaild `next` URL."""
```
Comment (emullaney) [inline tracpro/groups/tests/test_views.py]: spelling: invalid

File: tracpro/groups/tests/test_middleware.py (hunk 1 above)
Comment (emullaney) [inline tracpro/groups/tests/test_middleware.py]: actually, now that I re-read this comment, this is fine.

**Comment by emullaney** [comment]
Just a remaining Profile warning which can be fixed later, if needed, then :dart:

**Pull request merged**